Library/Homebrew/livecheck/strategy/gnu.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      module Gnu
+        module_function
+
+        NICE_NAME = "GNU"
+
+        PROJECT_NAME_REGEXES = [
+          %r{/(?:gnu|software)/(.+?)/},
+          %r{//(.+?)\.gnu\.org(?:/)?$},
+        ].freeze
+
+        def match?(url)
+          url.match?(%r{//.+?\.gnu\.org$|gnu\.org/(?:gnu|software)/}i) &&
+            !url.include?("savannah.")
+        end
+
+        def find_versions(url, regex = nil)
+          match_list = PROJECT_NAME_REGEXES.map { |r| url.match(r) }.compact
+          return { matches: {}, regex: regex, url: url } if match_list.blank?
+
+          odebug "\nMultiple project names found: #{match_list}\n" if match_list.length > 1
```

Comment:
Always favour `<<~EOS` over `\n` in strings.

Reply:
Just to make sure, this is what you're looking for here, correct?

```ruby
if match_list.length > 1
  odebug <<~EOS

    Multiple project names found: #{match_list}

  EOS
end
```

Reply:
Yup.